### PR TITLE
#153 In case 100 load increase fail we launch the scenarios for the m…

### PR DESCRIPTION
**Pull request by gautierbureau**
…aximum load increase passing.

**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (update of algorithms)
- [ ] main documentation was updated (update of input/output file, update of algorithms)
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: sources/Launcher/DYNMarginCalculationLauncher.cpp [hunk 1]
```diff
@@ -124,26 +124,43 @@ MarginCalculationLauncher::launch() {
...
   // We start with 100% as it is the most common result of margin calculations on real large cases
-  SimulationResult result100;
-  findOrLaunchLoadIncrease(loadIncrease, 100., marginCalculation->getAccuracy(), result100);
-  results_[idx].setStatus(result100.getStatus());
+  results_[idx].setStatus(resultMaxVariation.getStatus());
   std::vector<double > maximumVariationPassing(events.size(), 0.);
-  if (result100.getSuccess()) {
-    findAllLevelsBetween(0., 100., marginCalculation->getAccuracy(), allEvents, toRun);
+  if (resultMaxVariation.getSuccess()) {
+    findAllLevelsBetween(0., maxVariation, marginCalculation->getAccuracy(), allEvents, toRun);
```
Comment (rosiereflo) [inline sources/Launcher/DYNMarginCalculationLauncher.cpp:163]: Why not using minVariation here?

File: sources/Launcher/DYNMarginCalculationLauncher.cpp [hunk 2]
```diff
@@ -154,24 +171,29 @@ MarginCalculationLauncher::launch() {
...
       TraceInfo(logTag_) << DYNAlgorithmsLog(AlgorithmsWallTime, "Margin calculation", diff.total_milliseconds()/1000) << Trace::endline;
       TraceInfo(logTag_) << "============================================================ " << Trace::endline;
       cleanResultDirectories(events);
       return;
     }
-  }  // if the loadIncrease failed, nothing to do, the next algorithm will try to find the right load level
+  }
   TraceInfo(logTag_) << Trace::endline;
 
+  // we force the dichotomie between 0 and 100 event if 100 is not working
```
Comment (rosiereflo) [inline sources/Launcher/DYNMarginCalculationLauncher.cpp:190]: typo: even

**Review by rosiereflo — CHANGES_REQUESTED** [review]
Would it be possible to add a nrt ?

File: sources/Launcher/DYNMarginCalculationLauncher.cpp (hunk 1 above)
Comment (gautierbureau) [reply sources/Launcher/DYNMarginCalculationLauncher.cpp]: done

**Comment by gautierbureau** [comment]
> Would it be possible to add a nrt ?

Some nrt already have some issues with 100 and go through the new code, but those nrt have 0 failing and not 100 and 75 passing for example. I don't know if it is simple to create such case...

File: nrt/data/IEEE14/MC_file_div/reference/aggregatedResults.xml [hunk 3]
```diff
@@ -1,25 +1,7 @@
 <?xml version="1.0" encoding="ISO-8859-1" standalone="no"?>
 <aggregatedResults xmlns="http://www.rte-france.com/dynawo">
-  <loadIncreaseResults loadLevel="100" status="DIVERGENCE"/>
-  <loadIncreaseResults loadLevel="50" status="DIVERGENCE"/>
-  <loadIncreaseResults loadLevel="25" status="CONVERGENCE">
+  <loadIncreaseResults loadLevel="0" status="CONVERGENCE">
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply nrt/data/IEEE14/MC_file_div/reference/aggregatedResults.xml:3]: No as a scenario is not working at 0

File: nrt/data/IEEE14/MC_failing_load_increase/reference/aggregatedResults.xml [hunk 4]
```diff
@@ -1,7 +1,4 @@
 <?xml version="1.0" encoding="ISO-8859-1" standalone="no"?>
 <aggregatedResults xmlns="http://www.rte-france.com/dynawo">
-  <loadIncreaseResults loadLevel="100" status="CRITERIA_NON_RESPECTED"/>
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply nrt/data/IEEE14/MC_failing_load_increase/reference/aggregatedResults.xml:3]: Not anymore as 0 is not working

File: nrt/data/IEEE14/MC_file_div/reference/aggregatedResults.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (rosiereflo) [reply nrt/data/IEEE14/MC_file_div/reference/aggregatedResults.xml:3]: well we have to modify the test then as this tests an intermediate result

File: nrt/data/IEEE14/MC_failing_load_increase/reference/aggregatedResults.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (rosiereflo) [reply nrt/data/IEEE14/MC_failing_load_increase/reference/aggregatedResults.xml:3]: not sure that the platform is going to like it

[the remark this answers is not in the document]
Comment (gautierbureau) [reply nrt/data/IEEE14/MC_failing_load_increase/reference/aggregatedResults.xml:3]: I tested it and it works

File: nrt/data/IEEE14/MC_file_div/reference/aggregatedResults.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gautierbureau) [reply nrt/data/IEEE14/MC_file_div/reference/aggregatedResults.xml:3]: only thing I can do is add a new test case witj only the disconnect group event that would yield a margin of 15

[the remark this answers is not in the document]
Comment (rosiereflo) [reply nrt/data/IEEE14/MC_file_div/reference/aggregatedResults.xml:3]: why not

**Review by rosiereflo — APPROVED** [review]

**Comment by rosiereflo** [comment]
I wait before merging this one as I would like the on-going regression on the nrts to be analyzed before redefining the references

**Comment by gautierbureau** [comment]
> I wait before merging this one as I would like the on-going regression on the nrts to be analyzed before redefining the references

You mean regressions from another dev not this PR ?

**Comment by rosiereflo** [comment]
> > I wait before merging this one as I would like the on-going regression on the nrts to be analyzed before redefining the references
> 
> You mean regressions from another dev not this PR ?

correct

**Comment by rosiereflo** [comment]
converting to draft to avoid a merge before nrt are ok

**Comment by rosiereflo** [comment]
Could you please solve the conflict? The issue on dyn-algo was solved so this one can be merged

**Pull request merged**